salt/utils/jinja.py

```diff
@@ -206,6 +211,309 @@ def ensure_sequence_filter(data):
+
+
+def quote(txt):
+    '''
+    Wraps a text around quotes.
+
+    .. code-block:: jinja
+
+        {% set my_text = 'my_text' %}
+        {{ my_text | quote }}
+
+    will be rendered as:
+
+    .. code-block:: text
+
+        'my_text'
+    '''
+    return pipes.quote(txt)
+
+
+def regex_search(txt, rgx, ignorecase=False, multiline=False):
+    '''
+    Searches for a pattern in the text.
+
+    .. code-block:: jinja
+
+        {% set my_text = 'abcd' %}
+        {{ my_text | regex_search('^(.*)BC(.*)$', ignorecase=True) }}
+
+    will be rendered as:
+
+    .. code-block:: text
+
+        ('a', 'd')
+    '''
+    flag = 0
+    if ignorecase:
+        flag |= re.I
+    if multiline:
+        flag |= re.M
+    obj = re.search(rgx, txt, flag)
+    if not obj:
+        return
+    return obj.groups()
+
+
+def regex_match(txt, rgx, ignorecase=False, multiline=False):
+    '''
+    Searches for a pattern in the text.
+
+    .. code-block:: jinja
+
+        {% set my_text = 'abcd' %}
+        {{ my_text | regex_match('^(.*)BC(.*)$', ignorecase=True) }}
+
+    will be rendered as:
+
+    .. code-block:: text
+
+        ('a', 'd')
+    '''
+    flag = 0
+    if ignorecase:
+        flag |= re.I
+    if multiline:
+        flag |= re.M
+    obj = re.match(rgx, txt, flag)
+    if not obj:
+        return
+    return obj.groups()
+
+
+def regex_replace(txt, rgx, val, ignorecase=False, multiline=False):
+    r'''
+    Searches for a pattern and replaces with a sequence of characters.
+
+    .. code-block:: jinja
+
+        {% set my_text = 'lets replace spaces' %}
+        {{ my_text | regex_replace('\s+', '__') }}
+
+    will be rendered as:
+
+    .. code-block:: text
+
+        lets__replace__spaces
+    '''
+    flag = 0
+    if ignorecase:
+        flag |= re.I
+    if multiline:
+        flag |= re.M
+    compiled_rgx = re.compile(rgx, flag)
+    return compiled_rgx.sub(val, txt)
+
+
+def uuid_(val):
+    '''
+    Returns a UUID corresponding to the value passed as argument.
+
+    .. code-block:: jinja
+
+        {{ 'example' | uuid }}
+
+    will be rendered as:
+
+    .. code-block:: text
+
+        f4efeff8-c219-578a-bad7-3dc280612ec8
+    '''
+    return str(uuid.uuid5(GLOBAL_UUID, str(val)))
+
+
+### List-related filters
+
+
+def unique(lst):
+    '''
+    Removes duplicates from a list.
+
+    .. code-block:: jinja
+
+        {% set my_list = ['a', 'b', 'c', 'a', 'b'] -%}
+        {{ my_list | unique }}
+
+    will be rendered as:
+
+    .. code-block:: text
+
+        ['a', 'b', 'c']
+    '''
+    if not isinstance(lst, collections.Hashable):
+        return list(set(lst))
+    return lst
+
+
+def lst_min(lst):
+    '''
+    Returns the min value of a list.
+
+    .. code-block:: jinja
+
+        {% set my_list = [1,2,3,4] -%}
+        {{ my_list | min }}
+
+    will be rendered as:
+
+    .. code-block:: text
+
+        1
+    '''
+
+    if not isinstance(lst, collections.Hashable):
+        return float(min(lst))
+    return float(lst)
```

Comment:
Has this been tested to see what happens when we use this on a non-numeric type?